File: backend/app/schemas/reports.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel

from app.schemas.intelligence import DecisionPlanResponse


class SavedReportCreate(BaseModel):
    plan: DecisionPlanResponse


class SavedReportView(BaseModel):
    id: str
    source_report_id: str
    report_type: str
    title: str
    mime_type: str
    size_bytes: int
    provenance_count: int
    created_at: datetime


class SavedReportReady(BaseModel):
    report: SavedReportView
    download_url: str
    expires_at: datetime


class SignedDownloadResponse(BaseModel):
    url: str
    expires_at: datetime

File: frontend/types/reports.ts
export interface SavedReport {
  id: string;
  source_report_id: string;
  report_type: string;
  title: string;
  mime_type: string;
  size_bytes: number;
  provenance_count: number;
  created_at: string;
}

export interface SavedReportReady {
  report: SavedReport;
  download_url: string;
  expires_at: string;
}
